packages/flutter/lib/src/widgets/navigator_pop_handler.dart

```diff
@@ -0,0 +1,89 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'framework.dart';
+import 'navigator.dart';
+import 'notification_listener.dart';
+import 'pop_scope.dart';
+
+/// Enables the handling of system back gestures.
+///
+/// Typically wraps a nested [Navigator] widget and allows it to handle system
+/// back gestures in the [onPop] callback.
+///
+/// {@tool dartpad}
+/// This sample demonstrates how to use this widget to properly handle system
+/// back gestures when using nested [Navigator]s.
+///
+/// ** See code in examples/api/lib/widgets/navigator_pop_handler/navigator_pop_handler.0.dart **
+/// {@end-tool}
+///
+/// See also:
+///
+///  * [PopScope], which allows toggling the ability of a [Navigator] to
+///    handle pops.
+///  * [NavigationNotification], which indicates whether a [Navigator] in a
+///    subtree can handle pops.
+class NavigatorPopHandler extends StatefulWidget {
```

Comment:
Yeah, it's just a convenience for supporting back gestures when using a nested Navigator.

There was some discussion about this [in the design doc](https://docs.google.com/document/d/1BGCWy1_LRrXEB6qeqTAKlk-U2CZlKJ5xI97g45U7azk/edit#heading=h.wbk8nky7zicq).  Handling this scenario with WillPopScope is fairly concise compared to with PopScope, so I added this widget to make it simpler, since it seems to be a common use case.